com.github.crosswire.Xiphos.json

```diff
@@ -0,0 +1,111 @@
+        /* We need to access and store files */
+        "--filesystem=home",
+        /* Allow communication with network to get content */
+        "--share=network"
+    ],
+    "modules": [
+    {
+        "name": "libgsf",
+        "sources": [
+            {
+                "type": "archive",
+                "url": "https://ftp.gnome.org/pub/gnome/sources/libgsf/1.14/libgsf-1.14.41.tar.xz",
+                "sha256": "150b98586a1021d5c49b3d4d065d0aa3e3674ae31db131af5372499d2d3f08d3"
+            }
+        ]
+    },
+    {
+        "name": "gconf",
+        "config-opts": [ "--disable-static", "--disable-gtk-doc", "--disable-orbit",
+                            "--disable-introspection" ],
+        "sources": [
+            {
+                "type": "archive",
+                "url": "https://download.gnome.org/sources/GConf/3.2/GConf-3.2.6.tar.xz",
+                "sha256": "1912b91803ab09a5eed34d364bf09fe3a2a9c96751fde03a4e0cfa51a04d784c"
+            }
+        ]
+    },
+    {
+        "name": "biblesync",
+        "buildsystem": "cmake",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://github.com/karlkleinpaste/biblesync/releases/download/1.1.2/biblesync-1.1.2.tar.gz",
+                    "sha256": "886e1ac165b8900254c00a182fa34e3680ef3aa25051418470f15ea09ac02005"
+                }
+            ]
+    },
+    {
+        "name": "sword",
+        "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/bibletime/crosswire-sword-mirror.git",
+                    "commit": "311b60d114246dbe01ce82d1285ce3c2d98a6f4e"
+                }
+            ]
+    },
+    {
+        "name": "rarian",
+        "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://rarian.freedesktop.org/Releases/rarian-0.8.1.tar.bz2",
+                    "sha256": "aafe886d46e467eb3414e91fa9e42955bd4b618c3e19c42c773026b205a84577"
+                }
+            ]
+    },
+    {
+        "name": "gnome-doc-utils",
+        "sources": [
+                {
+                    "type": "archive",
+                    "url": "http://ftp.gnome.org/pub/GNOME/sources/gnome-doc-utils/0.20/gnome-doc-utils-0.20.10.tar.xz",
+                    "sha256": "cb0639ffa9550b6ddf3b62f3b1add92fb92ab4690d351f2353cffe668be8c4a6"
+                }
+            ]
+    },
+    {
+        "name": "libgtkhtml4",
+        "sources": [
+                {
+                    "type": "archive",
+                    "url": "http://ftp.gnome.org/pub/GNOME/sources/gtkhtml/4.10/gtkhtml-4.10.0.tar.xz",
+                    "sha256": "ca3b6424fb2c7ac5d9cb8fdafb69318fa2e825c9cf6ed17d1e38d9b29e5606c3"
+                }
+            ]
+    },
+    {
+        "name": "xiphos",
+        "buildsystem": "simple",
+        "build-commands": [
+            "./waf configure --prefix=/app --enable-webkit2 --gtk=3",
```

Comment:
Considering the build flags of gtk3 and webkit2, why is `libgtkhtml` built?

Reply:
Because it needs it! The app uses webkit2 for display and a choice of either webkit1(!) or gtkhtml for the editor component. I've got [manifests available](https://github.com/nedrichards/xiphos-flatpak) for basically every combination of build option, my view was that this was the best for security and flathub compliance but happy to take feedback.

Reply:
Is gtkhtml really the better choice? That is straight from 2009.

Reply:
This way you have webkit2 handling the untrusted input from the network and gtkhtml handling local notes the user has added, which, whilst a potential attack vector - should be a bit less likely.

I tried combining the webkit1 for editing and webkit2 but that basically moved everything to webkit1 - which seemed worse. This isn't something I feel particularly happy about either.

Reply:
Interesting. Might be worth discussing with the upstream developers?

Reply:
For sure. Would this be a blocker on flathub inclusion? Or just a "let's talk and see if we can get this better in a future release".

Reply:
Ah yes, I thought I remembered seeing this. This has been [reported upstream as well](https://github.com/crosswire/xiphos/issues/794), since Fedora has a WebKit1 deprecation, looks like there isn't a quick solution planned and we can either do this or go full WebKit1.

Reply:
Based on what you said I believe the current setup is ideal. It is not a blocker as you say I just want to ensure the projects are moving in the right direction and flathub isn't just a dump of "insecure things no distro would take".

Reply:
Totally agreed.